src/main/java/service/DukeService.java

```diff
@@ -0,0 +1,121 @@
+package service;
+
+import java.util.ArrayList;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+import exceptions.ServiceException;
+
+
+/**
+ * Core of Duke
+ */
+public class DukeService {
+    private ArrayList<Task> tasks;
+
+    /**
+     * Constructor
+     */
+    public DukeService() {
+        tasks = new ArrayList<>();
+    }
+
+    private String numberOfElementsAnnouncement() {
+        return String.format("You have %d elements in the list", tasks.size());
+    }
+
+    /**
+     * Adds a new task to taskList
+     * @param toAdd the new task
+     * @return a dedicated message
+     */
+    public DukeResponse addTask(Task toAdd) {
+        tasks.add(toAdd);
+        return new DukeResponse("Added: " + toAdd + "\n" + numberOfElementsAnnouncement() + "\n");
+    }
+
+    /**
+     * Initializes all tasks from text file
+     * @param otherTasks all the tasks
+     */
+    public void addInitializedTasks(ArrayList<Task> otherTasks) {
+        this.tasks.addAll(otherTasks);
+    }
+
+    /**
+     * Parse tasks to string to write to files
+     * @param parser parser
+     * @return a list of strings
+     */
+    public String[] getParsedTasks(Function<Task, String> parser) {
+        String[] results = new String[this.tasks.size()];
+        for (int i = 0; i < tasks.size(); i++) {
+            results[i] = parser.apply(tasks.get(i));
+        }
+        return results;
+    }
+
+    /**
+     * Deletes a tasks
+     * @param position position to be deleted
+     * @return a dedicated message
+     * @throws ServiceException if position is invalid
+     */
+    public DukeResponse deleteTask(int position) throws ServiceException {
+        position--;
+        if (position < 0 || position >= tasks.size()) {
+            throw new ServiceException("Position to delete is not valid :(");
+        }
+        Task toRemove = tasks.get(position);
+        tasks.remove(position);
+        StringBuilder sb = new StringBuilder();
+        sb.append("A task has been deleted: \n").append(toRemove).append("\n").append(numberOfElementsAnnouncement());
```

Comment:
I think this line is a bit too long as it exceeds 80 characters. Perhaps you could make it into 2 lines?